### PR TITLE
Backend Service and Data Layer Tests

### PR DESCRIPTION
**Pull request by MartynasBas**
Created tests for Services which covers all of it. 
Created tests for DataAccessLayer Filters and Models. Other parts could not be properly tested for their increased complexity and current time constraints.

### CODE REVIEW

**Comment by MartynasBas** [comment]
[Task 12: Cover remaining Services with Tests](https://trello.com/c/V6e6N4RK/131-task-12-cover-remaining-services-with-tests)

File: billchop-be/BillChopBETests/DataAccessLayer/Filters/BillDbFilterTests.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+            //Act
+            var result = filter.ApplyFilter(queryableBills).ToList();
+
+            //Assert
+            result.ShouldBe(bills);
+        }
+
+        [Test]
+        public void LoanDbFilter_WhenThereAreBillsWereCreatedBeforeCertainTime_ShouldReturnAllBillsCreatedBeforeCertainTime()
```
Comment (AinoraZ) [inline billchop-be/BillChopBETests/DataAccessLayer/Filters/BillDbFilterTests.cs]: WhenThereAreBillsThatWereCreatedBeforeCertainTime

File: billchop-be/BillChopBETests/DataAccessLayer/Models/UserTests.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            //Arrange    
+            var id = Guid.NewGuid();
+            var name = "test";
+            var email = "test@test.com";
+            var password = "password";
+            var groups = A.Fake<List<Group>>();
+            var loans = A.Fake<List<Loan>>();
+            var bills = A.Fake<List<Bill>>();
+            var paymentsMade = A.Fake<List<Payment>>();
+            var paymentsReceived = A.Fake<List<Payment>>();
```
Comment (AinoraZ) [inline billchop-be/BillChopBETests/DataAccessLayer/Models/UserTests.cs]: No need for fake lists, just replace with empty lists

File: billchop-be/BillChopBETests/DataAccessLayer/Models/UserWithTokenTests.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            //Arrange    
+            var id = Guid.NewGuid();
+            var name = "test";
+            var email = "test@test.com";
+            var password = "password";
+            var groups = A.Fake<List<Group>>();
+            var loans = A.Fake<List<Loan>>();
+            var bills = A.Fake<List<Bill>>();
+            var paymentsMade = A.Fake<List<Payment>>();
+            var paymentsReceived = A.Fake<List<Payment>>();
```
Comment (AinoraZ) [inline billchop-be/BillChopBETests/DataAccessLayer/Models/UserWithTokenTests.cs]: Same here with fake lists

File: billchop-be/BillChopBETests/DataAccessLayer/Models/GroupTests.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    class GroupTests
+    {
+        [Test]
+        public void Bill_WhenBillInfoIsPassed_ShouldReturnBillObject()
+        {
+            //Arrange    
+            var id = Guid.NewGuid();
+            var name = "Group one";
+            var users = A.Fake<List<User>>();
+            var bills = A.Fake<List<Bill>>();
```
Comment (AinoraZ) [inline billchop-be/BillChopBETests/DataAccessLayer/Models/GroupTests.cs]: No need for fakes here

File: billchop-be/BillChopBETests/DataAccessLayer/Models/BillTests.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        public void Bill_WhenBillInfoIsPassed_ShouldReturnBillObject()
+        {
+            //Arrange    
+            var id = Guid.NewGuid();
+            var name = "Shopping";
+            var total = 100;
+            var creationTime = DateTime.Now;
+            var loanerId = Guid.NewGuid();
+            var loaner = A.Fake<User>();
+            var loans = A.Fake<List<Loan>>();
```
Comment (AinoraZ) [inline billchop-be/BillChopBETests/DataAccessLayer/Models/BillTests.cs]: No need for fake list

File: billchop-be/BillChopBETests/DataAccessLayer/Filters/BillDbFilterTests.cs [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+            //Act
+            var result = filter.ApplyFilter(queryableBills).ToList();
+
+            //Assert
+            result.ShouldBe(bills);
+        }
+      
+        [Test]
+        public void LoanDbFilter_WhenThereAreBillsWereCreatedAfterCertainTime_ShouldReturnAllBillsCreatedAfterCertainTime()
```
Comment (AinoraZ) [inline billchop-be/BillChopBETests/DataAccessLayer/Filters/BillDbFilterTests.cs]: WhenThereAreBillsThatWereCreatedAfterCertainTime

**Review by AinoraZ — CHANGES_REQUESTED** [review]

**Review by AinoraZ — APPROVED** [review]
🚀

**Pull request merged**